crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -4,72 +4,74 @@ use ethrex_blockchain::{
     latest_canonical_block_hash,
     payload::{create_payload, BuildPayloadArgs},
 };
+use ethrex_core::types::{BlockHeader, ChainConfig};
 use serde_json::Value;
 use tracing::{info, warn};
 
 use crate::{
     types::{
-        fork_choice::{ForkChoiceResponse, ForkChoiceState, PayloadAttributesV3},
+        fork_choice::{ForkChoiceResponse, ForkChoiceState, PayloadAttributes},
         payload::PayloadStatus,
     },
     utils::RpcRequest,
     RpcApiContext, RpcErr, RpcHandler,
 };
 
 #[derive(Debug)]
-pub struct ForkChoiceUpdatedV3 {
+pub struct ForkChoiceUpdated {
     pub fork_choice_state: ForkChoiceState,
-    #[allow(unused)]
-    pub payload_attributes: Result<Option<PayloadAttributesV3>, String>,
+    pub payload_attributes: Result<Option<PayloadAttributes>, String>,
 }
 
-impl TryFrom<ForkChoiceUpdatedV3> for RpcRequest {
-    type Error = String;
-
-    fn try_from(val: ForkChoiceUpdatedV3) -> Result<Self, Self::Error> {
-        match val.payload_attributes {
-            Ok(attrs) => Ok(RpcRequest {
-                method: "engine_forkchoiceUpdatedV3".to_string(),
-                params: Some(vec![
-                    serde_json::json!(val.fork_choice_state),
-                    serde_json::json!(attrs),
-                ]),
-                ..Default::default()
-            }),
-            Err(err) => Err(err),
-        }
-    }
-}
-
-impl RpcHandler for ForkChoiceUpdatedV3 {
-    // TODO(#853): Allow fork choice to be executed even if fork choice updated v3 was not correctly parsed.
-    fn parse(params: &Option<Vec<Value>>) -> Result<Self, RpcErr> {
+impl ForkChoiceUpdated {
+    // TODO(#853): Allow fork choice to be executed even if fork choice updated was not correctly parsed.
+    fn parse(params: &Option<Vec<Value>>) -> Result<ForkChoiceUpdated, RpcErr> {
         let params = params
             .as_ref()
             .ok_or(RpcErr::BadParams("No params provided".to_owned()))?;
         if params.len() != 2 {
             return Err(RpcErr::BadParams("Expected 2 params".to_owned()));
         }
-        Ok(ForkChoiceUpdatedV3 {
+        Ok(ForkChoiceUpdated {
             fork_choice_state: serde_json::from_value(params[0].clone())?,
             payload_attributes: serde_json::from_value(params[1].clone())
                 .map_err(|e| e.to_string()),
         })
     }
 
-    fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
+    fn try_from(fork_choice_updated: &dyn ForkChoiceUpdatedImpl) -> Result<RpcRequest, String> {
+        let request = fork_choice_updated.request();
+        match &request.payload_attributes {
+            Ok(attrs) => Ok(RpcRequest {
+                method: fork_choice_updated.method(),
+                params: Some(vec![
+                    serde_json::json!(request.fork_choice_state),
+                    serde_json::json!(attrs),
+                ]),
+                ..Default::default()
```

Comment:
Nit: I'd prefer if we used some explicit fields here.

Reply:
I kind of prefer that too... but it's like [repeating arbitrary code](https://github.com/lambdaclass/ethrex/blob/9a2f8e0f86f1544aea25a6af66b34a3725b8a0c0/crates/networking/rpc/utils.rs#L190):

```rust
            id: RpcRequestId::Number(1),
            jsonrpc: "2.0".to_string(),
```